transaction/src/manifest/lexer.rs

```diff
@@ -364,13 +359,59 @@ impl Lexer {
     }
 
     fn unexpected_char(&self) -> LexerError {
-        LexerError::UnexpectedChar(self.text[self.current.full_index - 1], self.current)
+        let mut end = self.previous;
+        end.full_index += 1;
+
+        LexerError {
+            error_kind: LexerErrorKind::UnexpectedChar(self.text[self.current.full_index - 1]),
+            span: Span {
+                start: self.current,
```

Comment:
- removed `unexpected_char()`
- renamed `unexpected_char_previous()`  to `unexpected_char()`
- added `position` argument to `unexpected_char() since `self.previous` removed